python/pyspark/sql/connect/group.py

```diff
@@ -97,36 +97,46 @@ def agg(self, *exprs: Union[Column, Dict[str, str]]) -> "DataFrame":
             ),
             session=self._df._session,
         )
-        return res
 
     agg.__doc__ = PySparkGroupedData.agg.__doc__
 
-    def _map_cols_to_expression(self, fun: str, param: Union[Column, str]) -> Sequence[Column]:
-        return [
-            scalar_function(fun, col(param)) if isinstance(param, str) else param,
-        ]
+    def _numeric_agg(self, function: str, cols: Sequence[str]) -> "DataFrame":
+        from pyspark.sql.connect.dataframe import DataFrame
+
+        assert isinstance(function, str) and function in ["min", "max", "avg", "sum"]
+
+        assert isinstance(cols, list) and all(isinstance(c, str) for c in cols)
+
+        return DataFrame.withPlan(
```

Comment:
what happens when you only do this change but leave the `is_mumeric` out?

Reply:
the agg expressions are just literal strings, it will return something like:

```
In [6]: df.groupBy("age").agg(F.lit("max"), F.lit("col1")).show()
+---+---+----+
|age|max|col1|
+---+---+----+
|  2|max|col1|
|  5|max|col1|
+---+---+----+

```

Reply:
Yeah, but why not convert the function to a scalar function like we do for the above agg method?

Reply:
You're already validating the input anyways.

Reply:
Wait, sorry my bad.

Reply:
I see now why you do it that way, but I'm not convinced it's the best way. 

The only delta I see that can't be handled reasonably is the case of empty agg expressions where we know the function. And here I'm wondering if we should just get the schema instead.

Reply:
sound reasonable, we can analyze the schema in the client, then do not need to change the sever side. Let me take another look